Support for Velocity proxy

Krymonota opened the pull request:
This pull request adds support for [Velocity](https://www.velocitypowered.com/), a Minecraft server proxy with unparalleled server support, scalability, and flexibility. It's an alternative to BungeeCord, which is becoming increasingly popular. The Velocity API is somewhat similar to to that of BungeeCord, so porting was quite easy and maintaining it takes very little time. I tested this addition on my Velocity instance and everything works as expected.

Krymonota committed:
Add Velocity to project's readme file

Krymonota committed:
Add Velocity support

Krymonota committed:
Fix formatting in pom.xml

Krymonota committed:
Fix plugin message channel identifier

Krymonota committed:
Fix import of Subscribe annotation

Krymonota committed:
Fix error message showing up on logging

Velocity injects a SLF4J and no Java Util Logger

Krymonota committed:
Fix line indentation in Velocity classes

Krymonota committed:
Upgrade Velocity API version from 1.0.3 to 1.0.4

Rigner reviewed (commented):
Looks good, I'll test myself this plugin with Velocity to ensure everything is good

Also, I'll need to pass on the code to make sure our coding style (which you couldn't know), is properly set everywhere

Thanks ! :)

Krymonota committed:
Specify version (3.8.1) of maven-compiler-plugin

It's never a good idea not to specify the version, because then it
refers to the latest version that could break the build process when a
new version is released.

[WARNING] Some problems were encountered while building the effective
model for net.badlion:blccpsapivelocity:jar:1.2.1
[WARNING] 'build.plugins.plugin.version' for
org.apache.maven.plugins:maven-compiler-plugin is missing. @ line 24,
column 21
[WARNING]
[WARNING] It is highly recommended to fix these problems because they
threaten the stability of your build.

PSNRigner commented:
This project has been moved to https://github.com/BadlionClient/BadlionClientModAPI and Velocity support has been added.
